Sources/SKCore/FileBuildSettingsChange.swift

```diff
@@ -16,8 +16,16 @@ public enum FileBuildSettingsChange {
   /// The `BuildSystem` has no `FileBuildSettings` for the file.
   case removedOrUnavailable
 
-  /// The `FileBuildSettings` have been modified.
+  /// The `FileBuildSettings` have been modified or are newly available.
   case modified(FileBuildSettings)
+
+  /// The `BuildSystem` is providing fallback arguments which may not be correct,
+  /// or potentially is missing some generated inputs (e.g. prebuilding required).
+  ///
+  /// This indicates that diagnostics should be withheld for the given file. In addition, going from
```

Comment:
This is a policy decision that gets made in the code that uses the settings, not some kind of invariant of the build settings themselves.  I would remove this comment.

Reply:
With the above case in mind it isn't, we explicitly want to support settings not changing but the fallback state changing - it can mean new inputs are available.